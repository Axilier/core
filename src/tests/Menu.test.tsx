import React from 'react';
import { mount } from 'enzyme';
import { Menu } from '../index';

const menu = mount(
    <Menu open>
        <div>test menu</div>
    </Menu>,
);
describe('when the menu is used in an application', () => {
    it('should render the contained menu correctly', () => {
        expect(menu).toMatchSnapshot();
    });
    describe('when `open` is true', () => {
        it('should apply styles to show the menu', () => {
            expect(menu.childAt(0).props().style).toHaveProperty(
                'display',
                'flex',
            );
        });
    });
    describe('when `open` is false', () => {
        beforeAll(() => menu.setProps({ open: false }));
        it('should apply styles to not show the menu', () => {
            expect(menu.childAt(0).props().style).toHaveProperty(
                'display',
                'none',
            );
        });
    });
});
