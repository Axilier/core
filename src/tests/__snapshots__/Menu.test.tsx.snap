// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the menu is used in an application should render the contained menu correctly 1`] = `
<Menu
  open={true}
>
  <Layout
    className="container"
    orientation="column"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      className="container"
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <div
        className="main"
      >
        <div>
          test menu
        </div>
      </div>
    </div>
  </Layout>
</Menu>
`;
